barman/recovery_executor.py

```diff
@@ -1807,13 +1810,318 @@ def _backup_copy(self, backup_info, dest, remote_command=None, **kwargs):
+            raise DataTransferFailure.from_command_error("pg_combinebackup", e, msg)
+
+        self._end_message(synthetic_backup_info)
+
+        self.combine_end_time = datetime.datetime.now()
+        combine_time = total_seconds(self.combine_end_time - self.combine_start_time)
+        synthetic_backup_info.copy_stats = {
+            "combine_time": combine_time,
+        }
+
+        return synthetic_backup_info
+
+    def _backup_copy(
+        self,
+        backup_info,
+        dest,
+        tablespaces=None,
+        remote_command=None,
+        **kwargs,
+    ):
+        """
+        Perform the actual copy/move of the synthetic backup to destination
+
+        :param barman.infofile.SyntheticBackupInfo backup_info: the synthetic
+            backup info file
+        :param str dest: the destination directory
+        :param dict[str,str]|None tablespaces: a tablespace
+            name -> location map (for relocation)
+        :param str|None remote_command: default ``None``. The remote command to
+            recover the backup, in case of remote backup
+        """
+        # If it is a remote recovery we just follow the standard rsync copy process
+        if remote_command:
+            super(IncrementalRecoveryExecutor, self)._backup_copy(
+                backup_info, dest, tablespaces, remote_command, **kwargs
+            )
+            return
+        # If it is a local recovery we move the content from staging to destination
+        # Starts with tablespaces
+        if backup_info.tablespaces:
+            for tablespace in backup_info.tablespaces:
+                # By default a tablespace goes in the same location where
+                # it was on the source server when the backup was taken
+                destination = tablespace.location
+                # If a relocation has been requested for this tablespace
+                # use the user provided target directory
+                if tablespaces and tablespace.name in tablespaces:
+                    destination = tablespaces[tablespace.name]
+                # Move the content of the tablespace directory to destination directory
+                self._prepare_destination(destination)
+                tbs_source = backup_info.get_data_directory(
+                    tablespace_oid=tablespace.oid
+                )
+                self._move_to_destination(source=tbs_source, destination=destination)
+
+        # Then procede to move the content of the data directory
+        # We don't move the pg_tblspc as the _prepare_tablespaces method called earlier
+        # in the process already created this directory and required symlinks in the destination
+        data_source = backup_info.get_data_directory()
+        self._move_to_destination(
+            source=data_source, destination=dest, exclude_path_names={"pg_tblspc"}
+        )
+
+    def _move_to_destination(self, source, destination, exclude_path_names={}):
```

Comment:
```suggestion
    def _move_to_destination(self, source, destination, exclude_path_names=set()):
```

While `{"pg_tblspc"}` does create a `set`, `{}` creates an empty dict. Let's use `set()` to create an empty set properly.